imports/server/publications/puzzlesForPuzzleList.ts

```diff
@@ -0,0 +1,42 @@
+import { check, Match } from 'meteor/check';
+import MeteorUsers from '../../lib/models/MeteorUsers';
+import Puzzles from '../../lib/models/Puzzles';
+import Tags from '../../lib/models/Tags';
+import puzzlesForPuzzleList from '../../lib/publications/puzzlesForPuzzleList';
+import publishJoinedQuery from '../publishJoinedQuery';
+import definePublication from './definePublication';
+
+definePublication(puzzlesForPuzzleList, {
+  validate(arg) {
+    check(arg, {
+      huntId: String,
+      includeDeleted: Match.Optional(Boolean),
+    });
+    return arg;
+  },
+
+  async run({ huntId, includeDeleted = false }) {
+    if (!this.userId) {
+      return [];
+    }
+
+    const user = await MeteorUsers.findOneAsync(this.userId);
+    if (!user?.hunts?.includes(huntId)) {
+      return [];
+    }
+
+    publishJoinedQuery(this, {
+      model: Puzzles,
+      allowDeleted: includeDeleted,
+      foreignKeys: [{
+        field: 'tags',
+        join: {
+          model: Tags,
+        },
+      }],
```

Comment:
Oh, we actually need to make sure we're publishing all tags anywhere that we allow editing the tag list.